### ISSUE TITLE
setuptools / pkg_resources now also needs `platformdirs`

### ISSUE THREAD

**Issue by Athanasius** [py2exe/py2exe#175]
Yup, it's another run of the issue in https://github.com/py2exe/py2exe/issues/139 - this time it's `platformdirs`.

I'm working around it with a:
```diff
--- a/Build-exe-and-msi.py
+++ b/Build-exe-and-msi.py
@@ -84,6 +84,7 @@ OPTIONS = {
         'packages': [
             'asyncio',  # No longer auto as of py3.10+py2exe 0.11
             'multiprocessing',  # No longer auto as of py3.10+py2exe 0.11
+            'pkg_resources._vendor.platformdirs',  # Necessary 2023-01-17
             'sqlite3',  # Included for plugins
             'util',  # 2022-02-01 only imported in plugins/eddn.py
         ],
```
for now.

**Comment by Athanasius** [comment]
I've been puzzled as to why I only just noticed this, and I'd been doing test builds recently and the .exes ran fine.  infi.systray was last updated **years** ago, so it's not a change in *it* that's to blame.

But maybe the recent setuptools 66.0.0 update has caused this to come to light ?

**Comment by dimitry-unified-streaming** [comment]
This is specifically caused by https://github.com/pypa/setuptools/blob/main/CHANGES.rst#changes-1 :
* pypa/setuptools#3769: Replace 'appdirs' with 'platformdirs'.

E.g. for setuptools >= 66.0.0. Not sure how this should be implemented in py2exe, can it just be added to the list of packages in hooks.py, similar to 24b466871fc2d8cb61949bb26b168b3e921e8c36 ?

**Comment by albertosottile** [comment]
Saw this, looks like a quite easy fix in the hook (for now, until we fix the way `setuptools` vendors stuff permanently). However I am completely swamped at the moment, so I cannot work on this right away.

**Comment by albertosottile** [comment]
I wanted to work on "persistent" fix for this, but it looks like none of the tests in the CI can reproduce this (perhaps they do not require explicitly `pkg_resources` or any of its `extern` dependencies). 

Can you provide a minimal example that can be used to test this bug? Thanks

**Comment by Athanasius** [comment]
Unfortunately I am no longer part of EDMarketConnector development, so have a lot less interest in all things py2exe now.

And currently I can't get my old example at https://github.com/Athanasius/infi-systray_py2exe to even actually include infi.systray files in the library.zip.

The actual pkg_resources issue will be something to do with the:
```python
__import__("pkg_resources").declare_namespace(__name__)
from .traybar import SysTrayIcon
```
which is the entirety of `infi/systray/__init__.py` though.

**Comment by albertosottile** [comment]
> Unfortunately I am no longer part of EDMarketConnector development, so have a lot less interest in all things py2exe now. 

I am truly sorry to hear that.

Anyway, I attempted to fix this bug and you would find preliminary wheels with the fix here: https://github.com/py2exe/py2exe/actions/runs/4157014939.

In case you are still willing to test this, I would appreciate it.

**Comment by albertosottile** [comment]
Fixed in 0.13.0.1
